### PR TITLE
lobby fix: Adding instructions when we get a service error on conflicting joins

### PR DESCRIPTION
**Pull request by SamuelBellomo**
### Description (*)

When getting conflict error (if we forget to set -AuthProfile on two builds), we have zero feedback on this at the moment. Adding instructions to do this.

### Issue Number(s) (*)

Fixes issue(s): [MTT-2742](https://jira.unity3d.com/browse/MTT-2742)
### Manual testing scenarios

1. Removed ParrelSync automatic profile generation and tested main + clone joining.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs [hunk 1]
```diff
@@ -46,6 +50,24 @@ ClientGameNetPortal clientGameNetPortal
...
+                if ((error.OriginalException as LobbyServiceException).Reason == LobbyExceptionReason.LobbyConflict)
+                {
+                    // LobbyConflict can have multiple causes. Let's add other solutions here if there's other situations that arise for this.
+                    errorMessage += "\nSee logs for possible causes and solution.";
+                    Debug.LogError($"Got service error {error.Message} with LobbyConflict. Possible conflict cause: Trying to play with two builds on the " +
+                              $"same machine. Please use command line arg '{ClientMainMenuState.AuthProfileCommandLineArg} someName' to set a different auth profile.\n");
+                }
+            }
+
+            PopupPanel.ShowPopupPanel("Service error", errorMessage);
```
Comment (LPLafontaineB) [inline Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs:70]: Should this be moved to a separate script, like with ConnectionStatusMessageUIManager? Either way, that would also resolve [MTT-2734](https://jira.unity3d.com/browse/MTT-2734) I think

Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs:70]: Yeah you're right, I thought at first to make it lobby specific, but this applies to all service errors actually... let me look

Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs]: done

**Review by LPLafontaineB — DISMISSED** [review]

File: Assets/BossRoom/Scripts/Client/UI/UIServices.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+using System.Collections.Generic;
+using Unity.Multiplayer.Samples.BossRoom;
+using Unity.Multiplayer.Samples.BossRoom.Client;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Net.UnityServices.Infrastructure;
+using Unity.Multiplayer.Samples.BossRoom.Visual;
+using Unity.Services.Lobbies;
+using UnityEngine;
+
+public class UIServices : MonoBehaviour
```
[the remark this answers is not in the document]
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Client/UI/UIServices.cs]: for some reason, my Rider's config for code formatting got reset...
fixed

File: Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs [hunk 3]
```diff
@@ -1,8 +1,10 @@
+using System;
 using TMPro;
 using Unity.Multiplayer.Samples.BossRoom.Client;
 using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
 using Unity.Multiplayer.Samples.BossRoom.Shared.Net.UnityServices.Infrastructure;
 using Unity.Multiplayer.Samples.BossRoom.Shared.Net.UnityServices.Lobbies;
+using Unity.Services.Lobbies;
```
[the remark this answers is not in the document]
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs:7]: fixed

**Review by fernando-cortez — APPROVED** [review]

**Review by LPLafontaineB — APPROVED** [review]

**Review by LPLafontaineB — APPROVED** [review]

**Review by fernando-cortez — APPROVED** [review]

**Pull request merged**